src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc

```diff
@@ -526,6 +559,34 @@ void PX4FirmwarePlugin::guidedModeChangeAltitude(Vehicle* vehicle, double altitu
     }
 }
 
+void PX4FirmwarePlugin::guidedModeChangeGroundSpeed(Vehicle* vehicle, double groundspeed)
+{
+
+    vehicle->sendMavCommand(
```

Comment:
Nevertheless it could be also implemented like you did as a specialization of the FirmwarePlugin class. I dont know hat is prefered by the maintainers.

Reply:
@julled That's a good point, thanks for pointing it out. I think if both flight stacks interpret the mavlink message the same way (which ideally they should) then your proposal makes sense and introduces less code changes.
If however, the mavlink messages need to be filled differently due to different interpretations then I think the plugin way is nicer.
It's not net clear how APM does it so I'm going to keep it like this for now but I'm open to change it if we decide to move forward with APM as well.